src/stdatamodels/jwst/datamodels/util.py

```diff
@@ -151,6 +193,12 @@ def open(init=None, guess=True, **kwargs):  # noqa: A001
 
         return ModelContainer(init, **kwargs)
 
+    elif isinstance(init, (dict, asdf.AsdfFile)):
```

Comment:
I'm ok if we allow the `else` below to catch these cases (`dict`, `asdf.AsdfFile`). Having a special `if` for these to issue a slightly different error message doesn't seem worth the added code and tests.